src/test/scala/io/iohk/ethereum/network/PeerStatisticsSpec.scala

```diff
@@ -48,7 +48,9 @@ class PeerStatisticsSpec
 
     val statA = stats.stats(alice)
     statA.responsesReceived shouldBe 2
-    statA.lastSeenTimeMillis shouldBe >(statA.firstSeenTimeMillis)
+    // @dv: the corner case when responses are received simultaneously
+    // can ruin the tests for no reason
+    statA.lastSeenTimeMillis shouldBe >=(statA.firstSeenTimeMillis)
```

Comment:
Sorry for that. There may be better ways to test this via mocked clocks but I wanted to make sure that the first and last are treated separately. If they were always equal that would be an error.

Reply:
I have provided a clock instance, please take a look into it, the pull requests has grown bigger though :(